lib/ipam/ipam.go

```diff
@@ -676,67 +677,27 @@ func (c ipamClient) releaseIPsFromBlock(ctx context.Context, ips []net.IP, block
 			}
 		}
 
-		// Block exists - get the allocationBlock from the KVPair.
-		b := allocationBlock{obj.Value.(*model.AllocationBlock)}
+		// Determine whether or not the block's pool still matches the node.
+		released, err := c.maybeReleaseAffinity(ctx, obj.Value.(*model.AllocationBlock))
+		if err != nil {
```

Comment:
I think we need to handle the resource update conflict error type here - if we hit a conflict, we should start at the beginning of the loop again.